### PR TITLE
fix selfdestruct in bus-mapping

### PR DESCRIPTION
**Pull request by lispc**

### CODE REVIEW

File: bus-mapping/src/evm/opcodes.rs [hunk 1]
```diff
@@ -587,16 +587,21 @@ fn dummy_gen_selfdestruct_ops(
...
+        return Err(Error::AccountNotFound(sender));
+    }
+    let value = sender_account.balance;
     state.transfer(&mut exec_step, sender, receiver, value)?;
 
     if state.call()?.is_persistent {
         state.sdb.destruct_account(sender);
     }
 
+    state.handle_return(geth_step)?;
```
[the remark this answers is not in the document]
Comment (lispc) [reply bus-mapping/src/evm/opcodes.rs:605]: Thank you i will check this!

**Comment by lispc** [comment]
> not sure, but is `dummy_gen_selfdestruct_ops` still a good name or we can bump it to `gen_selfdestruct_ops` ?

I think keeping dummy is better. In prev convention with dummy it means circuit design is not finished yet so busmapping side we also think it is not finished completely

**Comment by lispc** [comment]
@adria0 i understand what you mean! I find another problem, currently `stop` opcode also calls `handle_return`. So I guess if we put a `STOP` as tx deployment constructor, it will panic too? 

Maybe in `handle_return`, we should also check opcode is `return` or `revert` (rather than stop or selfdestruct)?

**Comment by lispc** [comment]
thanks! so @adria0  do you think this pr makes sense, or it misses anything?

**Review by han0110 — APPROVED** [review]
All fixes look good to me, thanks for testing and fixing!

**Pull request merged**